### ISSUE TITLE
Example in docs for ByteToMessageDecoderVerifier.verifyDecoder is incorrect

### ISSUE THREAD

**Issue by Csloan7597** [apple/swift-nio#1243]
The example in [the docs for this function](https://apple.github.io/swift-nio/docs/current/NIOTestUtils/Enums/ByteToMessageDecoderVerifier.html#/s:12NIOTestUtils28ByteToMessageDecoderVerifierO06verifyF016inputOutputPairs14decoderFactoryySay3NIO0C6BufferV_Say10InboundOutQzGtG_xyctKAG0cdeF0RzSQAKRQlFZ) supplies an argument which does not match the function's signature. The example won't compile. 

> For ExampleDecoder that produces `ExampleDecoderOutput`s you would use this method the following way:

``` swift 
let expectedInOuts = [(exampleInput1, ExampleDecoderOutput("1")),
                      (exampleInput2, ExampleDecoderOutput("2"))
                     ]
XCTAssertNoThrow(
    try ByteToMessageDecoderVerifier.verifyDecoder(inputOutputPairs: expectedInOuts, decoderFactory: { ExampleDecoder() })
)
```

`expectedInOuts` is `[(ByteBuffer, ExampleDecoderOutput)]` when it should in fact be `[(ByteBuffer, [ExampleDecoderOutput])]`, as per the function signature:

``` swift
func verifyDecoder<Decoder: ByteToMessageDecoder>(inputOutputPairs: [(ByteBuffer, [Decoder.InboundOut])],
                                                                decoderFactory: @escaping () -> Decoder) throws where Decoder.InboundOut: Equatable
```

**Comment by weissi** [comment]
thank you for the report @Csloan7597

**Comment by PeterAdams-A** [comment]
I believe this was fixed in #1338
